src/blockdata/script.rs

```diff
@@ -239,9 +238,9 @@ pub fn read_scriptint(v: &[u8]) -> Result<i64, Error> {
 /// else as true", except that the overflow rules don't apply.
 #[inline]
 pub fn read_scriptbool(v: &[u8]) -> bool {
-    !(v.is_empty() ||
-        ((v[v.len() - 1] == 0 || v[v.len() - 1] == 0x80) &&
-         v.iter().rev().skip(1).all(|&w| w == 0)))
+    !(v.is_empty()
+      || ((v[v.len() - 1] == 0 || v[v.len() - 1] == 0x80)
+          && v.iter().rev().skip(1).all(|&w| w == 0)))
```

Comment:
Oh, at first I thought this is a closure. This whole chunk is super-hard to read. I believe the proper solution for such cases is to assign the values to clearly named variables and then have an inline expression (`!v.is_empty() || (var_a && car_b))

However note that `is_empty()` is actually bounds check for subsequent operations which makes it even uglier. This should help:
```rust
let last = math v.last() {
    Some(last) => last,
    None => return false,
};
// use `last` from now on
```

Reply:
Ha ha, its funny I didn't see that as a closure, soon as I started refactoring this code I dropped straight back into being a C programmer. The human brain is a mysterious thing.

Reply:
Please see: https://github.com/rust-bitcoin/rust-bitcoin/pull/808